apps/backend/src/modules/project-api-keys/project-api-key.service.ts

```diff
@@ -0,0 +1,130 @@
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { type Encryption } from "~/libs/modules/encryption/encryption.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Token } from "~/libs/modules/token/token.js";
+import { type Service } from "~/libs/types/types.js";
+import { type ProjectRepository } from "~/modules/projects/project.repository.js";
+
+import { ProjectApiKeyError } from "./libs/exceptions/exceptions.js";
+import {
+	type ProjectApiKeyCreateRequestDto,
+	type ProjectApiKeyCreateResponseDto,
+} from "./libs/types/types.js";
+import { ProjectApiKeyEntity } from "./project-api-key.entity.js";
+import { type ProjectApiKeyRepository } from "./project-api-key.repository.js";
+
+type Constructor = {
+	encryption: Encryption;
+	projectApiKeyRepository: ProjectApiKeyRepository;
+	projectRepository: ProjectRepository;
+	token: Token;
+};
+
+class ProjectApiKeyService implements Service {
+	private encryption: Encryption;
+
+	private projectApiKeyRepository: ProjectApiKeyRepository;
+
+	private projectRepository: ProjectRepository;
+
+	private token: Token;
+
+	public constructor({
+		encryption,
+		projectApiKeyRepository,
+		projectRepository,
+		token,
+	}: Constructor) {
+		this.projectApiKeyRepository = projectApiKeyRepository;
+		this.token = token;
+		this.projectRepository = projectRepository;
```

Comment:
we need to use project service, we shouldn't access other repositories inside this service
```suggestion
		this.projectService = projectService;
```

Reply:
Sure

Reply:
oh I see, than let's leave it